pkg/engine/plugins.go

```diff
@@ -262,25 +270,79 @@ func installPlugin(ctx context.Context, plugin workspace.PluginSpec) error {
 	logging.V(preparePluginVerboseLog).Infof(
 		"installPlugin(%s, %s): initiating download", plugin.Name, plugin.Version)
 
-	withProgress := func(stream io.ReadCloser, size int64) io.ReadCloser {
-		return workspace.ReadCloserProgressBar(stream, size, "Downloading plugin", cmdutil.GetGlobalColorization())
+	pluginID := fmt.Sprintf("%s-%s", plugin.Name, plugin.Version)
+	downloadMessage := "Downloading plugin " + pluginID
+
+	// Where possible, we'll use a progress bar to show download progress by
+	// wrapping the download stream with a progress reporting ReadCloser.
```

Comment:
I think this comment might be a little confusing. In both cases we have a progress bar, either the old style `ReadCloserProgressBar`, or the new style `NewProgressReportingCloser`, the difference being that NewProgressReportingCloser emits events, to be consumed by the display, and `ReadCloserProgressBar` directly renders a bar into the terminal (At least I think that's the difference?)